packages/vite/src/node/packages.ts

```diff
@@ -77,6 +77,24 @@ export function resolvePackageData(
   }
 
   const originalBasedir = basedir
+  const selfPackageData = findNearestPackageData(basedir, packageCache)
+  const isSelfReference =
+    selfPackageData?.data.exports && selfPackageData?.data.name === pkgName
```

Comment:
I think we should place it before this line.
https://github.com/vitejs/vite/blob/ee562070f2a46d8f71eac117333e950737e04bf1/packages/vite/src/node/plugins/resolve.ts#L735
Checking the self package isn't needed in other places calling `resolvePackageData`.